core/control.py

```diff
@@ -64,29 +65,45 @@ def perform_wavelength_sweep(self, wavelength_start: float, wavelength_end: floa
             print("-----Conducting laser sweep-----")
             print("Parameters:-----------")
             print("Laser start/stop/steps:", wavelength_start, wavelength_end, steps)
+            print("Reps:",reps)
+            print(f"Power meter averaging over {self.power_meter.get_average()} samples")
+            print(f"Power meter frequency set to {self.power_meter.get_wavelength()} nm")
             print("----------------------")
         for i, wavelength in enumerate(scan_range):
 
             self.laser.set_wavelength(wavelength)
-            time.sleep(3)
+            time.sleep(5) # <20 is okay for Quantifi
             if self.laser.wait_steady_state() == True:
-                power_readings[i] = self.power_meter.read()
+                time.sleep(2) #wait another 2 seconds
+                for j in range(reps):
+                    power_readings[j,i] = self.power_meter.read()
```

Comment:
Since you're already averaging when you read from the power meter isn't there some relationship between the number of samples taken and the error? like 1/sqrt(n) or something so you don't need to store different readings like this

Reply:
We can set the power meter to average over N samples and take 1 reading. But I am unaware of the method to get the mean/std from the power meter directly, if it exists. Hence asking the power meter to average over N samples, and we take some reps of data, so we can estimate a standard deviation.